Intersect.Server/Entities/Entity.cs

```diff
@@ -1950,6 +1950,45 @@ public virtual void TryAttack(Entity target)
                         this, Strings.Combat.addsymbol + (int) healthRecovered, CustomColors.Combat.Heal
                     );
                 }
+
+                var manasteal = thisPlayer.GetEquipmentBonusEffect(EffectType.Manasteal) / 100f;
+                var manaRecovered = Math.Min(enemyVitals[(int) Vitals.Mana], manasteal * baseDamage);
+
+                if (manaRecovered > 0) //Don't send any +0 msg's.
+                {
+                    AddVital(Vitals.Mana, (int) manaRecovered);
+                    PacketSender.SendActionMsg(
+                        this, Strings.Combat.addsymbol + (int) manaRecovered, CustomColors.Combat.AddMana
+                    );
+
+                    enemy.SubVital(Vitals.Mana, (int)manaRecovered);
+
+                    //Whether the attacking player should recover more mana than the enemy has
+                    //the rest will damage the enemy's health
+                    if (manasteal * baseDamage > enemyVitals[(int) Vitals.Mana])
+                    {
+                        var amountToRemove = Math.Floor(manasteal * baseDamage - manaRecovered);
+                        if(amountToRemove > 0)
+                        {
+                            enemy.SubVital(Vitals.Health, (int) amountToRemove);
+                            PacketSender.SendActionMsg(
+                                enemy, Strings.Combat.removesymbol + amountToRemove, CustomColors.Combat.TrueDamage
+                            );
+                        }
+                    }
+                }
+                else
+                {
+                    //If enemy has 0 mana, all value will be dealt as damage
```

Comment:
Even if it only deals damage, it should still recover mana

Also, because this is computed after life steal, you need to Math.Min() against the _current_ health remaining (which might not be what is stored in `enemyVitals` because lifesteal may have been subtracted already), and use that to compute mana recovered and damage dealt.